src/netlink_namespace.c

```diff
@@ -719,36 +730,35 @@ static int ir_init (void)
   DEBUG("netlink_namespace plugin: ir_init: Switching to namespace on file '%s'.", ns_name);
   if ((targetns = open(ns_name, O_RDONLY)) < 0) {
     ERROR("Unable to open namespace file '%s'", ns_name);
-    close_fds(&selfns, &targetns);
+    ir_init_error(&selfns, &targetns);
     return -1;
   }
 
   if (setns(targetns, CLONE_NEWNET) != 0) {
     ERROR("netlink_namespace plugin: ir_init: Unable to switch namespace to '%s'", ns_name);
-    close_fds(&selfns, &targetns);
-    return -1;
+    ir_init_error(&selfns, &targetns);
+    abort();
   }
 
   if (NULL == (nltarget = mnl_socket_open(NETLINK_ROUTE))) {
     ERROR("netlink_namespace plugin: ir_init: Unable to open socket to namespace '%s'", ns_name);
-    close_fds(&selfns, &targetns);
+    ir_init_error(&selfns, &targetns);
     return -1;
   }
 
   if (0 < mnl_socket_bind(nltarget, 0, MNL_SOCKET_AUTOPID)) {
     ERROR("netlink_namespace plugin: ir_init: Unable to bind to target namespace");
-    close_fds(&selfns, &targetns);
+    ir_init_error(&selfns, &targetns);
     return -1;
   }
 
   /* Get back to main namespace */
   if (setns(selfns, CLONE_NEWNET) != 0) {
     ERROR("netlink_namespace plugin: ir_init: Unable to get back to original namespace");
-    close_fds(&selfns, &targetns);
+    ir_init_error(&selfns, &targetns);
     return -1;
```

Comment:
This is actually where you want that abort() - we're trying to undo state, which will be actively harmful, and we failed to undo it. I cannot see any sensible thing to do here other than abort.